packages/fields/src/actions/set-as-homepage.tsx

```diff
@@ -0,0 +1,192 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import type { Settings, Page } from '@wordpress/core-data';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import type { Action, ActionModal } from '@wordpress/dataviews';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from './utils';
+import type { CoreDataError, PostWithPermissionsAndContext } from '../types';
+
+const PAGE_POST_TYPE = 'page';
+
+const SetAsHomepageModal: ActionModal< PostWithPermissionsAndContext >[ 'RenderModal' ] =
+	( { items, closeModal, onActionPerformed } ) => {
+		const [ item ] = items;
+		const pageTitle = getItemTitle( item );
+		const { currentHomePage, showOnFront } = useSelect( ( select ) => {
+			const { getEntityRecord } = select( coreStore );
+			const siteSettings: Settings | undefined = getEntityRecord(
+				'root',
+				'site'
+			);
+			const pageOnFront = siteSettings?.page_on_front;
+			return {
+				currentHomePage: getEntityRecord(
+					'postType',
+					'page',
+					pageOnFront
+				),
+				showOnFront: siteSettings?.show_on_front,
+			};
+		} );
+		const currentHomePageTitle = getItemTitle( currentHomePage as Page );
```

Comment:
I'm getting an error here due to the fact that `currentHomePage` is undefined (which results in `item.title` checks within this function to raise an error). This happens when the homepage is set to display the latest posts.

Looking at how other code works (in the [site editor](https://github.com/WordPress/gutenberg/blob/6d36c1b8ee1c3c68601a61959bc1c2602ee107d1/packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js#L38), for example), to find a homepage candidate we need to check that

1. `show_on_front` is `page`
2. `page_on_front` is not 0

Is that your understanding as well?

Reply:
Reading now the code below, it already considers the two situations (front is page or posts), so perhaps we can just move this line to where it's needed (within the sprintf).

Reply:
Thanks, good spot! I've moved this line to within the sprint, here: https://github.com/WordPress/gutenberg/pull/65426/files#diff-4320028e1b1fe30d612641c6c8fe5eab313d695872751b95faeb3d00ba490c91R125. I think this is working as I'm not seeing this error when the homepage is set to latest posts.